### PR TITLE
Feature/us9261 update cache refactor pinservice

### PR DESCRIPTION
**Pull request by Doug-Shannon**
removed AWS Cache editing
Refactored pin service-
groupService -> groupInquiryService
added pinCollectionProcessingService
added sayHiService
new linting rules engine
lots of other stuff

### CODE REVIEW

File: src/app/components/create-group/create-group-data.service.spec.ts [hunk 1]
```diff
@@ -9,325 +9,345 @@ import { CreateGroupService } from './create-group-data.service';
...
     });
+  });
 
-    it('should create the create group service',
-        inject([CreateGroupService], (s: CreateGroupService) => {
-            expect(s).toBeTruthy();
-        })
-    );
+  it(
+    'should create the create group service',
```
Comment (JoeKer1) [inline src/app/components/create-group/create-group-data.service.spec.ts]: Not sure if I like this prettier. But I get why it did it.

File: src/app/components/create-group/create-group-data.service.ts [hunk 2]
```diff
@@ -164,10 +205,10 @@ export class CreateGroupService {
...
 
   /*
-  * This will clear meeting day, meeting time, and meeting frequency
-  * if the group is flexible
-  * else it will format the meeting time data the way it needs to be for submission
-  */
+    * This will clear meeting day, meeting time, and meeting frequency
+    * if the group is flexible
+    * else it will format the meeting time data the way it needs to be for submission
+    */
```
Comment (plachmann) [inline src/app/components/create-group/create-group-data.service.ts]: Glad this comment is now properly formatted. :)

File: src/app/components/neighbors/neighbors.component.ts [hunk 3]
```diff
@@ -71,18 +79,22 @@ export class NeighborsComponent implements OnInit, OnDestroy {
 
   private processAndDisplaySearchResults(searchLocationString, searchKeywordString, lat, lng, filterString): void {
     // TODO: We can probably move these next three calls to be in pin service directly. But will cause more refactoring
-    this.pinSearchResults.pinSearchResults =
-      this.pinService.addNewPinToResultsIfNotUpdatedInAwsYet(this.pinSearchResults.pinSearchResults);
+    // this.pinSearchResults.pinSearchResults = this.pinService.addNewPinToResultsIfNotUpdatedInAwsYet(
+    //   this.pinSearchResults.pinSearchResults
+    // );
```
Comment (plachmann) [inline src/app/components/neighbors/neighbors.component.ts]: Do we want to go ahead and ditch this commented code?

File: src/app/components/create-group/create-group-data.service.ts [hunk 4]
```diff
@@ -8,10 +8,22 @@ import { LookupService } from '../../services/lookup.service';
...
+  GroupPaths,
+  groupPaths
+} from '../../shared/constants';
 import * as _ from 'lodash';
 import * as moment from 'moment';
+import { UtcTimeFormatPipe } from '../../pipes/utc-time-format.pipe';
+
+import { Router, ActivatedRoute } from '@angular/router';
+
+import {} from '../../shared/constants';
```
Comment (JoeKer1) [inline src/app/components/create-group/create-group-data.service.ts]: can delete

File: src/app/components/neighbors/neighbors.component.ts [hunk 5]
```diff
@@ -71,18 +79,22 @@ export class NeighborsComponent implements OnInit, OnDestroy {
...
-    this.pinSearchResults.pinSearchResults =
-      this.pinService.ensureUpdatedPinAddressIsDisplayed(this.pinSearchResults.pinSearchResults);
+    // this.pinSearchResults.pinSearchResults = this.pinService.ensureUpdatedPinAddressIsDisplayed(
+    //   this.pinSearchResults.pinSearchResults
+    // );
 
-    this.pinSearchResults.pinSearchResults =
-      this.pinService.sortPinsAndRemoveDuplicates(this.pinSearchResults.pinSearchResults);
+    this.pinSearchResults.pinSearchResults = this.pinCollectionProcessingService.sortPinsAndRemoveDuplicates(
+      this.pinSearchResults.pinSearchResults
```
Comment (JoeKer1) [inline src/app/components/neighbors/neighbors.component.ts]: Should we just do this as part of search in the .map of the Obs in pin-service? disregard this comment as I haven't even looked at that file yet.

File: src/app/components/neighbors/neighbors.component.ts [hunk 6]
```diff
@@ -71,18 +79,22 @@ export class NeighborsComponent implements OnInit, OnDestroy {
...
+    );
 
-    this.pinSearchResults.pinSearchResults =
-        this.pinService.removePinFromResultsIfDeleted(this.pinSearchResults.pinSearchResults,
-                                                      this.state.getDeletedPinIdentifier());
+    // this.pinSearchResults.pinSearchResults = this.pinService.removePinFromResultsIfDeleted(
```
Comment (JoeKer1) [inline src/app/components/neighbors/neighbors.component.ts]: more commented code

**Pull request merged**